### PR TITLE
Hipp 1048 manage my teams

### PR DESCRIPTION
**Pull request by rdawson-hmrc**

### CODE REVIEW

File: app/views/team/ManageMyTeamsView.scala.html [hunk 1]
```diff
@@ -36,11 +39,46 @@ <h1 class="govuk-heading-l">
...
     </div>
 
-
     <div class="call-out-panel govuk-!-margin-top-7 govuk-!-margin-bottom-7">
-        <p class="govuk-body">
-            @messages("manageMyTeams.noTeams")
-        </p>
+        <div class="call-out-flex-header">
+            <h2>@messages("manageMyTeams.yourTeams")</h2>
+            <a href="@controllers.team.routes.CreateTeamStartController.startCreateTeam()" class="govuk-button">@messages("manageMyTeams.addNewTeam")</a>
```
Comment (PaulCDurham) [inline app/views/team/ManageMyTeamsView.scala.html:45]: We should use the GovukButton class for this (we prefer code over raw HTML in digital). Look at the ManageTeamView for an example.

File: app/views/team/ManageMyTeamsView.scala.html [hunk 2]
```diff
@@ -36,11 +39,46 @@ <h1 class="govuk-heading-l">
...
+                    <tr class="govuk-table__row">
+                        <th scope="row" class="govuk-table__header">
+                            <a href="@controllers.team.routes.ManageTeamController.onPageLoad(team.id)" class="govuk-body govuk-link govuk-link--no-visited-state">@team.name</a>
+                        </th>
+                        <td class="govuk-table__cell">@team.teamMembers.size</td>
+                        <td class="govuk-table__cell utcDateTime">@team.created.format(DateTimeFormatter.ISO_DATE_TIME)</td>
+                    </tr>
+                }
+            </tbody>
+        </table>
```
Comment (PaulCDurham) [inline app/views/team/ManageMyTeamsView.scala.html:67]: Use GovukTable for tables. ManageTeamView has an example of that too.

File: conf/messages.en [hunk 3]
```diff
@@ -457,6 +457,11 @@ manageMyTeams.title = Manage teams
 manageMyTeams.heading = Manage teams
 manageMyTeams.hint = Teams can be used to help you manage your applications and create APIs.
 manageMyTeams.noTeams = You have not created any teams
```
Comment (PaulCDurham) [inline conf/messages.en:459]: If this isn't used any more could we remove it?

**Review by PaulCDurham — CHANGES_REQUESTED** [review]

File: app/views/team/ManageMyTeamsView.scala.html [hunk 4]
```diff
@@ -36,11 +39,46 @@ <h1 class="govuk-heading-l">
...
+                const date = new Date(elUtcDateTime.textContent),
+                    isValidDate = !isNaN(date.valueOf());
+                if (isValidDate) {
+                    elUtcDateTime.textContent = date.toLocaleDateString(undefined, {day: "numeric", month: "short", year: "numeric"});
+                } else {
+                    console.warn(`Expected a valid date/time string but found "${utcDateTime.textContent}"`);
+                }
+            });
+        });
+    </script>
```
[the remark this answers is not in the document]
Comment (rdawson-hmrc) [reply app/views/team/ManageMyTeamsView.scala.html:82]: I will raise a separate ticket for this as discussed

File: app/views/team/ManageMyTeamsView.scala.html (hunk 1 above)
Comment (rdawson-hmrc) [reply app/views/team/ManageMyTeamsView.scala.html:45]: done

File: app/views/team/ManageMyTeamsView.scala.html (hunk 2 above)
Comment (rdawson-hmrc) [reply app/views/team/ManageMyTeamsView.scala.html:67]: done

File: conf/messages.en (hunk 3 above)
Comment (rdawson-hmrc) [reply conf/messages.en:459]: message now reinstated as discussed so leaving it in

File: app/controllers/team/ManageMyTeamsController.scala [hunk 5]
```diff
@@ -20,19 +20,25 @@ import com.google.inject.{Inject, Singleton}
...
   view: ManageMyTeamsView
-) extends FrontendBaseController with I18nSupport {
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad(): Action[AnyContent] = identify {
+  def onPageLoad(): Action[AnyContent] = identify.async {
     implicit request =>
-      Ok(view(request.user))
+      service.findTeams(request.user.email).map(_ match {
+        case Seq() => Redirect(controllers.routes.UnauthorisedController.onPageLoad.url)
```
[the remark this answers is not in the document]
Comment (rdawson-hmrc) [reply app/controllers/team/ManageMyTeamsController.scala:40]: done

File: app/controllers/team/ManageMyTeamsController.scala [hunk 6]
```diff
@@ -20,19 +20,25 @@ import com.google.inject.{Inject, Singleton}
...
+  service: ApiHubService,
   view: ManageMyTeamsView
-) extends FrontendBaseController with I18nSupport {
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  def onPageLoad(): Action[AnyContent] = identify {
+  def onPageLoad(): Action[AnyContent] = identify.async {
     implicit request =>
-      Ok(view(request.user))
+      service.findTeams(request.user.email).map(_ match {
```
[the remark this answers is not in the document]
Comment (rdawson-hmrc) [reply app/controllers/team/ManageMyTeamsController.scala:39]: done - how does a user sign in without an email address? both of the Sign In options say to use one

[the remark this answers is not in the document]
Comment (PaulCDurham) [reply app/controllers/team/ManageMyTeamsController.scala:39]: That's stub sign in. In production we sign in with either LDAP creds or magically via Stride. In both cases the user's email address is defined as an optional data item we can retrieve during the sign-in process.

File: app/views/team/ManageMyTeamsView.scala.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (rdawson-hmrc) [reply app/views/team/ManageMyTeamsView.scala.html:82]: Added to app.js with scope for adding other formatting options

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**